### PR TITLE
Adding transaction for Zuul tests

### PR DESCRIPTION
**Pull request by jsanemet**
- Adding a transaction that allows Cibyl to fetch test cases from a Zuul build.
- The transaction allows to filter tests by:
  - Name
  - Result
  - Duration, as a range.
- The test response provides access to all the data Cibyl needs to present.
- Unit testing filtering and data processing.

### CODE REVIEW

**Review by cescgina — APPROVED** [review]

File: cibyl/sources/zuul/transactions/__init__.py [hunk 1]
```diff
@@ -358,6 +365,87 @@ def get(self):
...
+    def with_name(self, *pattern: str) -> 'TestsRequest':
+        """Will limit request to tests whose name follows a certain pattern.
+
+        :param pattern: Regex pattern for the desired test name.
+        :return: The request's instance.
+        """
+
+        def test(response):
+            return any(
+                matches_regex(patt, response.name) for patt in pattern
```
Comment (adrianfusco) [inline cibyl/sources/zuul/transactions/__init__.py:391]: Maybe `for pattern in patterns`?

File: cibyl/sources/zuul/transactions/__init__.py [hunk 2]
```diff
@@ -358,6 +365,87 @@ def get(self):
...
+    def with_status(self, *status: TestStatus) -> 'TestsRequest':
+        """Will limit request to tests on a certain status.
+
+        :param status: Desired status of the test cases.
+        :return: The request's instance.
+        """
+
+        def test(response):
+            return any(
+                response.status == patt for patt in status
```
Comment (adrianfusco) [inline cibyl/sources/zuul/transactions/__init__.py:406]: Maybe the same here.

**Review by adrianfusco — APPROVED** [review]
Just two simple comments. The PR looks really good.

File: cibyl/sources/zuul/transactions/__init__.py (hunk 1 above)
Comment (jsanemet) [reply cibyl/sources/zuul/transactions/__init__.py:391]: I saw somewhere to name * variables in singular, like in: *pattern. That decision has led me to this thing here. The whole file uses this convention, so I am not going to bother changing such thing. I will most likely not do it again next time though.

Comment (adrianfusco) [reply cibyl/sources/zuul/transactions/__init__.py:391]: If you're not going to do it anymore then let's comment it with the team. At this way all of we have the same pattern.

Thank you, Jorge.

**Pull request merged**